### ISSUE TITLE
The Game Changer 

### ISSUE THREAD

**Issue by kemong** [instabot-py/instabot.py#227]
This is probably the game changer for Instabot
![works](https://cloud.githubusercontent.com/assets/22748697/19336227/e86ee35e-9133-11e6-88f1-8b1310c665bd.png)

First : I make my method modular. I separate my modified method into small files, so It works with the original Instabot.py with small changes.
![files](https://cloud.githubusercontent.com/assets/22748697/19336297/47b0ece0-9134-11e6-824f-40a376979913.png)

Second : In this example.py you can CHOOSE which mode you want to execute with some detail explanation there
![bot](https://cloud.githubusercontent.com/assets/22748697/19336294/4470011a-9134-11e6-9d17-eb0bd57228f2.png)

Third : You can still use the original Instabot.py as usual.

From now, i'll make my modified method in separate file. So i suggest everybody use the same approach to. Just to make the instabot.py more clear and unbloated.

Cheers :beer: 
Just check my pull request @LevPasha

**Comment by virginsince1997** [comment]
@kemong i got this error trying to run

Traceback (most recent call last):
  File "C:\Users\abbey\Desktop\instabot.py-master\example.py", line 4, in <module>
    from instabot import InstaBot
  File "C:\Users\abbey\Desktop\instabot.py-master\instabot.py", line 718
    print current_user
                     ^
SyntaxError: invalid syntax

**Comment by kemong** [comment]
i've update the pull request, but you can try to use this :

[instabot.py.zip](https://github.com/LevPasha/instabot.py/files/526188/instabot.py.zip)

**Comment by virginsince1997** [comment]
@kemong , thank you

**Comment by dave240170** [comment]
What are the stats how many extra followers a day do you get?

**Comment by virginsince1997** [comment]
Most of the followers I get are from My feeds, As for me, I get up to 60 a
day, I run it very early in the morning.

**Comment by arsischeon** [comment]
So @kemong your new Mods are only follow&unfollow features right? It doesn't do the liking/commenting at the same time? Thanks for your contribution!

**Comment by Bossishere** [comment]
Mode 5 isnt working for more than 25 unfollows in my case. Why?

**Comment by dave240170** [comment]
In previous testing, it will 403 error if you go too quick, it needs to be
spread out over the hour. You will get a 12 hour or 24 hour ban for going
to quick or worse.

**Comment by Bossishere** [comment]
This is what happens when i use Mode = 5 after about 5 minutes

 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: xqueenmillionx
peopleinlove : Get media id on recent feed 
 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: engineer_ali_emam
peopleinlove : Get media id on recent feed 
 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: alvcoaching
peopleinlove : Get media id on recent feed 
 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: wowasv777

**Comment by Bossishere** [comment]
This is what happens if i use mode =5 it tries to unfollow but doesnt do it... Why? Pls Help

 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: xqueenmillionx
xxxxxx : Get media id on recent feed 
 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: engineer_ali_emam
xxxxxx : Get media id on recent feed 
 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: alvcoaching
xxxxxx : Get media id on recent feed 
 15.10.2016 01:38
Media in recent feed = 24
Trying to unfollow: wowasv777

**Comment by drewhga** [comment]
There is a limit line in unfollow_protocol.py which limits the number you
can unfollow

1. Edit unfollow_protocol.py using IDLE

2. Change limit in line 13 to the max number of unfollows per day

Example :::
From
limit = random.randint(10,22) +1

To
limit = 999999999999999999999999

That should solve your problem

**Comment by arsischeon** [comment]
@kemong One more thing, on Mode 1 it often freezes on the stage 'Checking user info...'. It just stays there forever until I manually abort the run and restart. Any idea on where to check to solve this? Thanks!

**Comment by kemong** [comment]
@Bossishere , Yes @drewhga  was right. I limit my unfollow counter.
To unlimited unfollow people, make the mode 5 run forever, like while True: 
Or you can use example.py here 
[instabot.py.zip](https://github.com/LevPasha/instabot.py/files/530827/instabot.py.zip)

@arsischeon Sorry, i can't solve that bug. Because the bug only appears occasionally. I can't figure it out (yet). 

For all of you who use Linux / Bash Shell. You can use this shell script to automatically kill the python bot after running for few hours, wait a few minutes and start the bot again. 

![sh](https://cloud.githubusercontent.com/assets/22748697/19408118/b58b3980-92df-11e6-966f-3838abfe581f.png)


For example, you start the bot in 7:00 am and want the bot to run untill 12:00 
Change the sleep time from 3600 to 14400
![sh1](https://cloud.githubusercontent.com/assets/22748697/19408121/cb9c5d26-92df-11e6-9c22-547cbed438ef.png)

And if you want to start it again after 2 hours break, change the sleep time from 300 to 7200
![sh2](https://cloud.githubusercontent.com/assets/22748697/19408126/012d702e-92e0-11e6-99de-c356d7cda725.png)

Don't forget to change the location of your bot files. Edit this shell script using any text editor.
Basically this script is intended to let the bots run for one hour, after one hour the shell with kill all my python bots (3 clients here) and restart them again 10 minutes later. So if one of my clients get some runtime exception, the shell guarantees me that it will restart the python bot. I can leave the shell script to handle my bots without having to login to my bot server so often just for restarting the bot which encounter occasional exception (eg: if the bot meets people with 0 following, it will raise exception. because i can't divide follower number with zero).
[bot.sh.zip](https://github.com/LevPasha/instabot.py/files/530828/bot.sh.zip)

**Comment by Bossishere** [comment]
Thank you but still isnt working because he stopps unfollowing and just tries to...

**Comment by ChuyuHsu** [comment]
I merged @kemong's PR and now it is working smoothly.
May I issue a PR which merges @kemong's PR?

**Comment by kemong** [comment]
@LevPasha someone ask you.. ^

**Comment by LevPasha** [comment]
@kemong @ChuyuHsu you can make PR, but same structure what i create (src folder). Then in example.py default mode should be 0. Last PR from @kemong have many bugs and it bad idea to make his mode default... Also i create this bot for python 3+, not for 3-.

**Comment by ChuyuHsu** [comment]
Since @LevPasha have already merge the modes @kemong, I think another PR is not necessary

**Comment by alexcheninfo** [comment]
@kemong Awesome. One question, which of the modes prevents the bot from following and unfollowing the same user? Or at least, minimizes the risk? I already got four accounts banned because of this behavior, ha.

**Comment by marcoamesty** [comment]
Hello @kemong @LevPasha. Thank you for your effort, the bot is really helpful.

Is there any mod to follow all followers of a specific user? Instead of following by tags.

**Comment by drewhga** [comment]
#171

**Comment by andreaswa123** [comment]
Hey @marcoamesty @drewhga ,

have you figured out any solution to the problem?
"Is there any mod to follow all followers of a specific user? Instead of following by tags."

Would be highly interested as well!

Best,
A
